File: devops/instrumentation/app/course/internal/course/course.go
package course

import "time"

type Course struct {
	ID            string    `json:"id" gorm:"type:char(36);primaryKey"`
	Code          string    `json:"code" gorm:"type:varchar(20);unique;not null"`
	Name          string    `json:"name" gorm:"type:varchar(100);not null"`
	Price         float64   `json:"price" gorm:"type:decimal(10,2);not null"`
	StartDate     time.Time `json:"start_date" gorm:"type:date;not null"`
	EndDate       time.Time `json:"end_date" gorm:"type:date;not null"`
	SeatAvailable int       `json:"seat_available" gorm:"type:int;not null"`
	CreatedAt     time.Time `json:"created_at" gorm:"autoCreateTime"`
	UpdatedAt     time.Time `json:"updated_at" gorm:"autoUpdateTime"`
}

type CourseCodeRequest struct {
	Code string `json:"code"`
}

File: tmp/old/devops/instrumentation/java-stack/app/course/src/main/java/com/example/course/domain/Course.java
package com.example.course.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.UuidGenerator;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "Courses")
@EntityListeners(AuditingEntityListener.class)
@Getter
@Setter
public class Course {

    @Id
    @Column(nullable = false, updatable = false)
    @GeneratedValue
    @UuidGenerator
    private UUID id;

    @Column(nullable = false, unique = true, length = 20)
    private String code;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private BigDecimal price;

    @Column(nullable = false)
    private LocalDate startDate;

    @Column(nullable = false)
    private LocalDate endDate;

    @Column(nullable = false)
    private int seat;

    @Column(nullable = false)
    private int seatAvailable;

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @LastModifiedDate
    @Column(nullable = false)
    private OffsetDateTime updatedAt;
}
